src/site/content/en/how-to-optimize-inp/dom-size-and-interactivity/index.md

```diff
@@ -0,0 +1,164 @@
+
+```html
+<div>
+  <div>
+    <div>
+      <div>
+        <!-- Contents -->
+      </div>
+    </div>
+  </div>
+</div>
+```
+
+When you see patterns like this, you can probably simplify them by flattening your DOM structure. Doing so will reduce the number of DOM elements, and likely give you an opportunity to simplify page styles.
+
+DOM depth may also be a symptom of the frameworks you use. In particular, component-based frameworks—such as those that rely on [JSX](https://reactjs.org/docs/introducing-jsx.html)—require you to nest multiple components in a parent container.
+
+However, many frameworks allow you to avoid nesting components by using what are known as fragments. Component-based frameworks that offer fragments as a feature include (but are not limited to) the following:
+
+- [React](https://reactjs.org/docs/fragments.html)
+- [Preact](https://preactjs.com/guide/v10/components/#fragments)
+- [Vue](https://v3-migration.vuejs.org/new/fragments.html)
+- [Svelte](https://svelte.dev/tutorial/svelte-fragment)
+
+By using fragments in your framework of choice, you can reduce DOM depth. If you're concerned about the impact flattening DOM structure has on styling, you might benefit from using more modern (and faster) layout modes such as [flexbox](https://developer.mozilla.org/docs/Learn/CSS/CSS_layout/Flexbox) or [grid](https://developer.mozilla.org/docs/Web/CSS/CSS_Grid_Layout).
+
+## Other strategies to consider
+
+Even if you take pains to flatten your DOM tree and remove unnecessary HTML elements to keep your DOM as small as possible, it can still be quite large and kick off a lot of rendering work as it changes in response to user interactions. If you find yourself in this position, there are some other strategies you can consider to limit rendering work.
+
+### Consider an additive approach
+
+You might be in a position where large parts of your page aren't initially visible to the user when it first renders. This could be an opportunity to lazy load HTML by omitting those parts of the DOM on startup, but add them in when the user interacts with the parts of the page that require the initially hidden aspects of the page.
+
+This approach is useful both during the initial load and perhaps even afterwards. For the initial page load, you're taking on less rendering work up front, meaning that your initial HTML payload will be lighter, and will render more quickly. This will give interactions during that crucial period more opportunities to run with less competition for the main thread's attention.
+
+If you have many parts of the page that are initially hidden on load, it could also speed up other interactions that trigger re-rendering work. However, as other interactions add more to the DOM, rendering work will increase as the DOM grows throughout the page lifecycle.
```

Comment:
There's probably a good argument to be made against this. Adding DOM elements triggers rendering work. So does removing them. It's arguably a best practice to leave those DOM elements in place once added, as any event listeners bound to DOM nodes will have to be re-established if they're added to the DOM again later.

Reply:
Fair point. I guess it depends on how many you're adding and when.

E.g. doing a soft nav on an SPA is almost certainly a good case to add AND remove (or just replace) elements, rather than just add. Scrolling down an infinite scroll or Twitter feed maybe is a case to only keep some elements in the DOM. But yeah agree it's tricky.

Happy to leave as is.